ctru-rs/src/mii.rs

```diff
@@ -0,0 +1,479 @@
+            color: partial_vec_to_u8_with_reverse(&raw_mouth_details[6..8]),
+            scale: partial_vec_to_u8_with_reverse(&raw_mouth_details[9..12]),
+            y_scale: partial_vec_to_u8_with_reverse(&raw_mouth_details[13..15]),
+        };
+
+        let mustache_details = MustacheDetails {
+            mouth_y_position: partial_vec_to_u8_with_reverse(&raw_mustache_details[0..4]),
+            mustache_style: partial_vec_to_u8_with_reverse(&raw_mustache_details[5..7]),
+        };
+
+        let beard_details = BeardDetails {
+            style: partial_vec_to_u8_with_reverse(&raw_beard_details[0..2]),
+            color: partial_vec_to_u8_with_reverse(&raw_beard_details[3..5]),
+            scale: partial_vec_to_u8_with_reverse(&raw_beard_details[6..9]),
+            y_position: partial_vec_to_u8_with_reverse(&raw_beard_details[10..14]),
+        };
+
+        let glass_details = GlassDetails {
+            style: partial_vec_to_u8_with_reverse(&raw_glass_details[0..3]),
+            color: partial_vec_to_u8_with_reverse(&raw_glass_details[4..6]),
+            scale: partial_vec_to_u8_with_reverse(&raw_glass_details[7..10]),
+            y_position: partial_vec_to_u8_with_reverse(&raw_glass_details[11..15]),
+        };
+
+        let mole_details = MoleDetails {
+            is_enabled: raw_mole_details[0],
+            scale: partial_vec_to_u8_with_reverse(&raw_mole_details[1..4]),
+            x_position: partial_vec_to_u8_with_reverse(&raw_mole_details[5..9]),
+            y_position: partial_vec_to_u8_with_reverse(&raw_mole_details[10..14]),
+        };
+
+        MiiData {
+            mii_options: options,
+            mii_selector_position: position,
+            mii_console_identity: device,
+            system_id,
+            mac_address: creator_mac,
+            mii_details: details,
+            mii_name,
+            height,
+            width,
+            face_style,
+            face_details,
+            hair_style,
+            hair_details,
+            eye_details,
+            eyebrow_details,
+            nose_details,
+            mouth_details,
+            mustache_details,
+            beard_details,
+            glass_details,
+            mole_details,
+            author_name,
+        }
+    }
+}
+
+// Methods to handle "_bits_", ``bitvec`` cannot compile to 32-bit targets, so I had to create a few
+// helper methods
+
+/// Transforms a u8 into a [bool; 8]
+fn vec_bit(data: u8) -> [bool; 8] {
+    (0..8)
+        .map(|i| (data & (1 << i)) != 0)
+        .collect::<Vec<bool>>()
+        .try_into()
+        .unwrap()
+}
+
+/// Transforms a [bool; 8] into an u8
+fn vec_bit_to_u8(data: [bool; 8]) -> u8 {
+    let mut result: u8 = 0;
+    data.map(u8::from).iter().for_each(|&bit| {
+        result <<= 1;
+        result ^= bit;
+    });
+    result
+}
+
+/// The reverse allows to write things on a more _humane_ way, but return a LE u8
+fn partial_vec_to_u8_with_reverse(data: &[bool]) -> u8 {
```

Comment:
This function's functionality is still unclear to me, maybe a better name and more descriptive docs would help?

Reply:
Maybe I need help writing the docs for this function, so I'll write what's the purpose of this functions here so you'll be able to give some suggestions.
The 3DS is a LE system, but for humans it's _easier_ to read and understand BE bits. So, this function is given a BE sequence of bits (e.g. ``[true, true, false]``) that gets: (i) filled with _``false``s_ to fill the remaining space until 8 bits (``u8``) are filled (e.g. ``[false, false, false, false, false, true, true, false]``), and (ii) the bits returned are transformed to be LE (e.g. ``[false, true, true, false, false, false, false, false]`` so the 3DS can understand them.

``partial_vec`` -> A Vec\<bool> (_in reality it is a slice_) that doesn't fill the 8 bits of an ``u8``
``with_reverse`` -> Where the values being returned are LE (""_reversed_"")

Maybe something like ``partial_bits_to_le_u8`` is better?

Reply:
Alright, that function was _actually_ misused. The purpose of the function was writing BE bits, filling them until the length was 8 bits and making them LE.

This function was used this way before, but now the data is directly from the 3DS and this isn't needed + it caused some _big big_ bugs. I've fixed it and documented it